frontend/catalyst/pennylane_extensions.py

```diff
@@ -151,20 +156,24 @@ def __init__(self, fn, *, method, h, argnum):
         self.h = h
         self.argnum = argnum
         if self.method != "fd":
```

Comment:
Slight simplification now:
```suggestion
        if self.method != "fd" and not isinstance(self.fn, qml.QNode):
```